src/garage/torch/optimizers/conjugate_gradient_optimizer.py

```diff
@@ -178,6 +180,42 @@ def step(self, f_loss, f_constraint):  # pylint: disable=arguments-differ
         self._backtracking_line_search(params, descent_step, f_loss,
                                        f_constraint)
 
+    @property
+    def state(self):
+        """dict: The hyper-parameters of the optimizer."""
+        return {
+            'max_constraint_value': self._max_constraint_value,
+            'cg_iters': self._cg_iters,
+            'max_backtracks': self._max_backtracks,
+            'backtrack_ratio': self._backtrack_ratio,
+            'hvp_reg_coeff': self._hvp_reg_coeff,
+            'accept_violation': self._accept_violation,
+        }
+
+    @state.setter
+    def state(self, state):
+        self._max_constraint_value = state.get('max_constraint_value', 0.2)
+        self._cg_iters = state.get('cg_iters', 10)
+        self._max_backtracks = state.get('max_backtracks', 15)
+        self._backtrack_ratio = state.get('backtrack_ratio', 0.8)
+        self._hvp_reg_coeff = state.get('hvp_reg_coeff', 1e-5)
+        self._accept_violations = state.get('accept_violations', False)
+
+    def __setstate__(self, state):
+        """Restore the optimizer state.
+
+        Args:
+            state (dict): State dictionary.
+
+        """
+        if 'hvp_reg_coeff' not in state['state']:
+            warnings.warn(
+                'Resuming ConjugateGradientOptimizer from corrupt state')
```

Comment:
hmm can you tell me more about why it's corrupt and what (if anything) i can do about it? this is just scary.

Reply:
This should only happen if you're resuming from a checkpoint made with a version of garage before this change. I'll write some tests to make sure that's the case.

Reply:
Okay, I've made it clearer what can be done to fix this problem, and made it a little less scary sounding.